session-replacement/src/main/java/com/amadeus/session/repository/redis/JedisSessionRepositoryFactory.java

```diff
@@ -57,17 +65,43 @@ static JedisPoolConfig configurePool(RedisConfiguration config) {
   }
 
   private RedisFacade singleInstance(JedisPoolConfig poolConfig, RedisConfiguration config) {
+    logger.debug("Connecting to Single Instance Redis cache....");
     int port = Integer.parseInt(config.port);
     String[] serverAndPort = config.server.split(":");
     if (serverAndPort.length > 1) {
       port = Integer.parseInt(serverAndPort[1]);
     }
-    return new JedisPoolFacade(new JedisPool(poolConfig, config.server, port, config.timeout));
+
+    SSLSocketFactory sslSocketFactory = null;
+    SSLParameters sslParameters = null;
+    HostnameVerifier hostnameVerifier = null;
+
+    if (config.useSSL){
+        sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+        sslParameters = new SSLParameters();
+        sslParameters.setEndpointIdentificationAlgorithm("HTTPS");
+        if(config.tls != null) {
+            sslParameters.setProtocols(config.tls);
+        } else {
+            //Enable all TLS (including weak TLSv1 and v1.1)
+            sslParameters.setProtocols(new String[]{"TLSv1", "TLSv1.1", "TLSv1.2"});
+        }
+    }
+
+    logger.debug("with following - values poolConfig: " + poolConfig.toString() + ", " +
```

Comment:
how about using "{}" placeholders e.g. `logger.debug("... poolConfig: {}...", poolConfig, ...);`
It reduces object creation if debug is not active.